.emacs.d/lisp/package/packup.el

```diff
@@ -9,17 +9,199 @@
+    (save-excursion
+      (let ((beg (region-beginning))
+            (end (region-end)))
+        (kotct/packup-mark-packages-in-region
+         (progn (goto-char beg) (line-beginning-position))
+         (progn (goto-char end) (line-beginning-position))))))
+   (t
+    (let ((inhibit-read-only t))
+      (kotct/packup-repeat-over-lines
+       (prefix-numeric-value arg)
+       (lambda ()
+         (forward-char 1)
+         (delete-char 1)
+         (insert kotct/packup-marker-char)))))))
+
+(defun kotct/packup-unmark (arg &optional interactive)
+  "Unmark the package for update at point in the Packup buffer.
+If a region is selected, unmark all the packages in the region.
+If an prefix-arg is passed unmark ARG times."
+  (interactive (list current-prefix-arg))
+  (let ((kotct/packup-marker-char ?\040))
+    (kotct/packup-mark arg interactive)))
+
+(defun kotct/packup-mark-all ()
+  (interactive)
+  (save-excursion
+    (kotct/packup-mark-packages-in-region (point-min) (point-max))))
+
+
+(defun kotct/packup-unmark-all ()
+    (interactive)
+    (save-excursion
+      (let ((kotct/packup-marker-char ?\040))
+        (kotct/packup-mark-packages-in-region (point-min) (point-max)))))
+
+(defun kotct/packup-initialize-buffer-contents ()
+  (let ((inhibit-read-only t))
+    (kill-region (point-min) (point-max)))
+  (package-refresh-contents)
+  (let ((install-list nil))
+    (dolist (package kotct/dependency-list)
+      (if (or (not (package-installed-p package))
+              (not (kotct/package-up-to-date-p package)))
+          (apply 'kotct/packup-insert-package-row (list package (package-desc-version (cadr (assq package package-alist)))))))))
+
+(defun kotct/packup-initialize-buffer ()
+  "Initializes the packup buffer."
+  (kill-all-local-variables)
+  (use-local-map packup-mode-map) ;; mapppppps
+  (setq major-mode 'packup-mode
+        mode-name "Packup"
+        buffer-read-only t ;; read only
+        mode-line-buffer-identification "packup")
+  (kotct/packup-initialize-buffer-contents))
+
+(defun kotct/packup-refresh ()
+  "Refreshes packages in current buffer."
+  (interactive)
+  (if (eq major-mode 'packup-mode)
+    (let ((inhibit-read-only t))
+      (erase-buffer)
+      (kotct/packup-initialize-buffer-contents))))
+
+(defun kotct/packup-help ()
+  (interactive)
+  (message "g-refresh m-mark u-unmark x-execute ?-help"))
+
+(defvar packup-mode-map
+  (let ((map (make-keymap)))
+    (define-key map "m" 'kotct/packup-mark)
+    (define-key map "M" 'kotct/packup-mark-all)
+    (define-key map "u" 'kotct/packup-unmark)
+    (define-key map "U" 'kotct/packup-unmark-all)
+    (define-key map "x" 'kotct/packup-do-update)
+    (define-key map "g" 'kotct/packup-refresh)
+    (define-key map "?" 'kotct/packup-help)
+    map))
+
+;;;###autoload
+(defun kotct/packup ()
+  "Creates an interactive buffer to install/update packages."
+  (interactive)
+  (let ((old-buf (current-buffer))
+        (buffer (get-buffer-create "*packup*")))
```

Comment:
<img width="589" alt="screen shot 2017-08-25 at 10 31 27 am" src="https://user-images.githubusercontent.com/1566689/29720789-99e613e2-8980-11e7-98b5-400665995d90.png">

<img width="587" alt="screen shot 2017-08-25 at 10 32 06 am" src="https://user-images.githubusercontent.com/1566689/29720809-af9fa2d4-8980-11e7-9bae-783292a538b0.png">

I think the display problems are due to capitalization.  Can this be capitalized?

Reply:
Actually, never mind.  It's not.  It seems to be a problem with the mode.

Reply:
No, I revise this again.  It's because you set `mode-line-buffer-identification` to `packup` that we get rid of the bolded name.